pkg/resourcecollector/persistentvolume.go

```diff
@@ -90,10 +92,43 @@ func (r *ResourceCollector) pvToBeCollected(
 func (r *ResourceCollector) preparePVResourceForCollection(
 	object runtime.Unstructured,
 ) error {
+	var pv v1.PersistentVolume
+	var currentSc string
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.UnstructuredContent(), &pv); err != nil {
+		return err
+	}
+	// Some time pv spec does not contains the storage class.
+	// In that case, we will get it from pvc spec.
+	if len(pv.Spec.StorageClassName) == 0 {
+		pvcName := pv.Spec.ClaimRef.Name
+		pvcNamespace := pv.Spec.ClaimRef.Namespace
+		pvc, err := r.coreOps.GetPersistentVolumeClaim(pvcName, pvcNamespace)
+		if err != nil {
+			return err
+		}
+		currentSc, err = utils.GetStorageClassNameForPVC(pvc)
```

Comment:
Instead of error'ing out can we skip adding the annotation?
There is a case where a PV can be manually bind mounted to a volume (backend storage volume) and a PVC w/o requiring a storage class.

Reply:
Fixed